artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederatedQueueConsumerImpl.java

```diff
@@ -144,6 +152,8 @@ private void connect() throws Exception {
 
    @Override
    public void close() {
+      started.set(false);
```

Comment:
The connect could be already scheduled and not yet executed while a disconnect could be scheduled with zero delay. In this case, the disconnect would be executed before a connect.